src/app/query/stacks/sip10/sip10-tokens.utils.ts

```diff
@@ -3,33 +3,29 @@ import type { CryptoAssetBalance, Sip10CryptoAssetInfo } from '@leather-wallet/m
 
 import { isUndefined } from '@shared/utils';
 
-import { getTicker } from '@app/common/utils';
+import { getStacksContractIdStringParts } from '@app/common/stacks-utils';
+import { getPrincipalFromContractId, getTicker } from '@app/common/utils';
 import type { SwapAsset } from '@app/query/common/alex-sdk/alex-sdk.hooks';
-import { getAssetStringParts } from '@app/ui/utils/get-asset-string-parts';
 
 export function isTransferableSip10Token(token: Partial<FtMetadataResponse>) {
   return !isUndefined(token.decimals) && !isUndefined(token.name) && !isUndefined(token.symbol);
 }
 
 export function createSip10CryptoAssetInfo(
-  contractId: string,
   key: string,
   ftAsset: FtMetadataResponse
 ): Sip10CryptoAssetInfo {
-  const { assetName, contractName, address } = getAssetStringParts(key);
-  const tokenName = ftAsset.name ? ftAsset.name : assetName;
+  const { contractAssetName } = getStacksContractIdStringParts(key);
+  const name = ftAsset.name ? ftAsset.name : contractAssetName;
 
   return {
     canTransfer: isTransferableSip10Token(ftAsset),
-    contractAddress: address,
-    contractAssetName: assetName,
-    contractId,
-    contractName,
+    contractId: key,
```

Comment:
This was the my mistake/bug here, I believed the contractId was the principal, which is incorrect. It is now set to the entire asset identifier string (fullyQualifiedName) so it can be used in parts where needed.